tracing-appender/src/rolling.rs

```diff
@@ -4,12 +4,57 @@ use std::fmt::Debug;
 use std::io;
 use std::path::Path;
 
+/// A file appender with the ability to rotate log files at a fixed schedule.
+///
+/// This struct implements [`std:io::Write` trait][write] and will block on any write operation.
+/// It can be used in conjunction with [`NonBlocking`][non-blocking] to perform writes without
+/// blocking the main thread.
+///
+/// [write]: https://doc.rust-lang.org/nightly/std/io/trait.Write.html
+/// [non-blocking]: ../non_blocking/struct.NonBlocking.html
+///
+/// # Examples
+///
+/// ```rust
+/// # fn docs() {
+/// use tracing_appender::rolling::RollingFileAppender;
+/// use tracing_appender::rolling::Rotation;
+/// use tracing_appender::non_blocking::NonBlocking;
+///
+/// let file_appender = RollingFileAppender::new(Rotation::HOURLY, "/some/directory", "prefix.log");
+/// let (non_blocking, _guard) = NonBlocking::new(file_appender);
```

Comment:
nit, take it or leave it: how about we show off the more concise verbose APIs by default? 
```suggestion
/// let file_appender = tracing_appender::rolling::hourly("/some/directory", "prefix.log");
/// let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
```